modules/platform/video/nuxeo-platform-video/src/main/java/org/nuxeo/ecm/platform/video/service/RecomputeVideoConversionsAction.java

```diff
@@ -0,0 +1,181 @@
+
+        protected List<String> conversionNames;
+
+        protected VideoService videoService = Framework.getService(VideoService.class);
+
+        public RecomputeRenditionsComputation() {
+            super(ACTION_FULL_NAME);
+        }
+
+        @Override
+        public void startBucket(String bucketKey) {
+            BulkCommand command = getCurrentCommand();
+            xpath = command.getParam(PARAM_XPATH);
+            conversionNames = command.getParam(PARAM_CONVERSION_NAMES);
+            if (conversionNames.isEmpty()) {
+                conversionNames = videoService.getAvailableVideoConversions()
+                                              .stream()
+                                              .map(VideoConversion::getName)
+                                              .collect(Collectors.toList());
+            }
+        }
+
+        @Override
+        protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+            log.debug("Compute action: {} generating conversions: {} for doc ids: {}", ACTION_NAME, conversionNames,
+                    ids);
+            for (String docId : ids) {
+                if (!session.exists(new IdRef(docId))) {
+                    log.debug("Doc id doesn't exist: {}", docId);
+                    continue;
+                }
+
+                DocumentModel workingDocument = session.getDocument(new IdRef(docId));
+                Property fileProp = workingDocument.getProperty(xpath);
+                if (!((fileProp instanceof Blob) || (fileProp instanceof BlobProperty))) {
+                    log.warn("Property: {} of doc id: {} is not a blob.", xpath, docId);
+                    continue;
+                }
+                Blob blob = (Blob) fileProp.getValue();
+                if (blob == null) {
+                    // do nothing
+                    log.debug("No blob for doc: {}", workingDocument.getId());
+                    continue;
+                }
+                try {
+                    VideoDocument videoDoc = workingDocument.getAdapter(VideoDocument.class);
+                    Video video = videoDoc.getVideo();
+
+                    for (String conversion : conversionNames) {
+                        // here we want to commit the transaction since the video conversion can be very long
+                        TransactionHelper.commitOrRollbackTransaction();
+                        TranscodedVideo transcodedVideo = null;
+                        try {
+                            transcodedVideo = videoService.convert(video, conversion);
+                        } catch (ConversionException e) {
+                            log.warn("Conversion: {} of doc id: {} has failed", conversion, docId);
+                        }
+                        TransactionHelper.startTransaction();
+                        saveRendition(session, new IdRef(workingDocument.getId()), conversion, transcodedVideo);
```

Comment:
`new IdRef(workingDocument.getId())` -> `workingDocument.getRef()` but it's actually even simpler to use the local variable I suggested before.